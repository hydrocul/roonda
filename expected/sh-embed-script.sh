cat $ROONDA_TMP_PATH/data.js | $ROONDA_SELF_PATH --v1 --json-to-python2
cat $ROONDA_TMP_PATH/data.rd | $ROONDA_SELF_PATH --v1 --sexpr-to-python2
cat $ROONDA_TMP_PATH/data.js | $ROONDA_SELF_PATH --v1 --json-to-python2 --replace-tag ROONDA_STDIN_DATA --output-code $ROONDA_TMP_PATH/script.py
cat $ROONDA_TMP_PATH/data.js | $ROONDA_SELF_PATH --v1 --json-to-python2 --replace-tag ROONDA_STDIN_DATA $ROONDA_TMP_PATH/script.py
cat $ROONDA_TMP_PATH/data.js | $ROONDA_SELF_PATH --v1 --json-to-python2 $ROONDA_TMP_PATH/script2.py
cat $ROONDA_TMP_PATH/data.js | $ROONDA_SELF_PATH --v1 --json-to-python2
cat $ROONDA_TMP_PATH/data.js | $ROONDA_SELF_PATH --v1 --json-to-python2 $ROONDA_TMP_PATH/script2.py
cat $ROONDA_TMP_PATH/data.js | $ROONDA_SELF_PATH --v1 --json-to-python2 $ROONDA_TMP_PATH/roonda_9f408e8e0470f4ed1bd165e6e9aaf32855509991.py
cat $ROONDA_TMP_PATH/data.js | $ROONDA_SELF_PATH --v1 --from-json --to-sexpr-obj
cat $ROONDA_TMP_PATH/data.js | $ROONDA_SELF_PATH --v1 --json-to-sexpr

#################################################
# data.js:
#################################################
# [["abc", 10],
#  ["def", 30],
#  ["ghi", 100]]

#################################################
# data.rd:
#################################################
# ("abc" 10)
# ("def" 30)
# ("ghi" 100)

#################################################
# script.py:
#################################################
# print(ROONDA_STDIN_DATA)

#################################################
# script2.py:
#################################################
# print(stdin_data)

#################################################
# roonda_9f408e8e0470f4ed1bd165e6e9aaf32855509991.py:
#################################################
# import sys
# 
# sys.stdout.write(str(stdin_data))
# sys.stdout.write(str("\n"))
