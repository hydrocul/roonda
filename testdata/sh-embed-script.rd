(sh v1)

(pipe (cat data.js)
      (roonda --v1 --json-to-python2))

(pipe (cat data.rd)
      (roonda --v1 --sexpr-to-python2))

(pipe (cat data.js)
      (roonda --v1 --json-to-python2 --replace-tag ROONDA_STDIN_DATA --output-code script.py))

(pipe (cat data.js)
      (roonda --v1 --json-to-python2 --replace-tag ROONDA_STDIN_DATA script.py))

(pipe (cat data.js)
      (roonda --v1 --json-to-python2 script2.py))

(pipe (cat data.js)
      (roonda json-to-python2))

(pipe (cat data.js)
      (roonda json-to-python2 script2.py))

(pipe (cat data.js)
      (roonda json-to-python2 ()
       (print stdin_data)
       (print "\n")))

(pipe (cat data.js)
      (roonda --v1 --from-json --to-sexpr-obj))

(pipe (cat data.js)
      (roonda json-to-sexpr))


<< data.js
[["abc", 10],
 ["def", 30],
 ["ghi", 100]]
data.js >>

<< data.rd
("abc" 10)
("def" 30)
("ghi" 100)
data.rd >>

<< script.py
print(ROONDA_STDIN_DATA)
script.py >>

<< script2.py
print(stdin_data)
script2.py >>

