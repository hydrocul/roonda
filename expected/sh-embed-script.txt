[["abc",10],["def",30],["ghi",100]]
[["abc",10],["def",30],["ghi",100]]
print([["abc",10],["def",30],["ghi",100]])
[['abc', 10], ['def', 30], ['ghi', 100]]
[['abc', 10], ['def', 30], ['ghi', 100]]
[["abc",10],["def",30],["ghi",100]]
[['abc', 10], ['def', 30], ['ghi', 100]]
[['abc', 10], ['def', 30], ['ghi', 100]]
(("abc" 10) ("def" 30) ("ghi" 100))
(("abc" 10) ("def" 30) ("ghi" 100))
